### PR TITLE
Implement functional color variables 4 [ButtonDanger, ButtonOutline, ButtonPrimary, ButtonClose]

### PR DESCRIPTION
**Pull request by VanAnderson**
This PR updates the following components to use functional color variables in preparation for color mode support:

-  ButtonDanger
-  ButtonOutline
-  ButtonPrimary
-  ButtonClose

Part of https://github.com/github/design-systems/issues/1219

### CODE REVIEW

**Review by bscofield — APPROVED** [review]
👍

File: src/Button/ButtonOutline.tsx [hunk 1]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    color: ${get('colors.btn.outline.hoverText')};
+    background-color: ${get('colors.btn.outline.hoverBg')};
+    border-color: ${get('colors.btn.outline.hoverBorder')};
+    box-shadow: ${get('shadows.btn.outline.hoverShadow')};
   }
   // focus must come before :active so that the active box shadow overrides
   &:focus {
     border-color: transparent;
-    box-shadow: ${get('buttons.outline.shadow.focus')};
+    box-shadow: ${get('colors.btn.outline.focusShadow')};
```
Comment (colebemis) [inline src/Button/ButtonOutline.tsx:22]:
```suggestion
    box-shadow: ${get('shadows..btn.outline.focusShadow')};
```

File: src/Button/ButtonOutline.tsx [hunk 2]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
 
   &:active {
-    color: ${get('buttons.outline.color.active')};
-    background-color: ${get('buttons.outline.bg.active')};
-    border-color: ${get('buttons.outline.border.active')};
-    box-shadow: ${get('buttons.outline.shadow.active')};
+    color: ${get('colors.btn.outline.selectedText')};
+    background-color: ${get('colors.btn.outline.selectedBg')};
+    box-shadow: ${get('shadows.btn.outline.selectedShadow')};
+    border-color: ${get('shadows.btn.outline.selectedBorder')};
```
Comment (colebemis) [inline src/Button/ButtonOutline.tsx:29]:
```suggestion
    border-color: ${get('colors.btn.outline.selectedBorder')};
```

File: src/Button/ButtonOutline.tsx [hunk 3]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
   }
 
   &:disabled {
-    color: ${get('buttons.outline.color.disabled')};
-    border-color: ${get('buttons.outline.border.default')};
-    background-color: ${get('buttons.outline.bg.disabled')};
+    color: ${get('colors.btn.outline.disabledText')};
+    background-color: ${get('buttons.outline.disabledBg')};
+    border: ${get('buttons.outline.disabledBorder')};
```
Comment (colebemis) [inline src/Button/ButtonOutline.tsx:35]:
```suggestion
    border-color: ${get('colors.btn.border')};
```

File: src/Button/ButtonOutline.tsx [hunk 4]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
-    border-color: ${get('buttons.outline.border.hover')};
-    box-shadow: ${get('buttons.outline.shadow.hover')};
+    color: ${get('colors.btn.outline.hoverText')};
+    background-color: ${get('colors.btn.outline.hoverBg')};
+    border-color: ${get('colors.btn.outline.hoverBorder')};
+    box-shadow: ${get('shadows.btn.outline.hoverShadow')};
   }
   // focus must come before :active so that the active box shadow overrides
   &:focus {
     border-color: transparent;
```
Comment (colebemis) [inline src/Button/ButtonOutline.tsx:21]:
```suggestion
    border-color: ${get('colors.btn.outline.focusBorder')};
```

File: src/Button/ButtonPrimary.tsx [hunk 5]
```diff
@@ -5,32 +5,34 @@ import {ComponentProps} from '../utils/types'
...
-    background-color: ${get('buttons.primary.bg.hover')};
-    border-color: ${get('buttons.primary.border.hover')};
+    color: ${get('colors.btn.primary.hoverText')};
+    background-color: ${get('colors.btn.primary.hoverBg')};
+    border-color: ${get('colors.btn.primary.hoverBorder')};
+    box-shadow: ${get('shadows.btn.primary.hoverShadow')};
   }
   // focus must come before :active so that the active box shadow overrides
   &:focus {
     border-color: transparent;
```
Comment (colebemis) [inline src/Button/ButtonPrimary.tsx:21]:
```suggestion
    border-color: ${get('colors.btn.primary.focusBorder')};
```

File: src/Button/ButtonPrimary.tsx [hunk 6]
```diff
@@ -5,32 +5,34 @@ import {ComponentProps} from '../utils/types'
...
-    box-shadow: ${get('buttons.primary.shadow.focus')};
-    background-color: ${get('buttons.primary.bg.focus')};
+    box-shadow: ${get('colors.btn.primary.focusShadow')};
```
Comment (colebemis) [inline src/Button/ButtonPrimary.tsx:22]:
```suggestion
    box-shadow: ${get('shadows.btn.primary.focusShadow')};
```

File: src/Button/ButtonPrimary.tsx [hunk 7]
```diff
@@ -5,32 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    border-color: ${get('shadows.btn.primary.selectedBorder')};
   }
 
   &:disabled {
-    color: ${get('buttons.primary.color.disabled')};
-    background-color: ${get('buttons.primary.bg.disabled')};
-    border-color: ${get('buttons.primary.border.disabled')};
+    color: ${get('colors.btn.primary.disabledText')};
+    background-color: ${get('buttons.primary.disabledBg')};
+    border: ${get('buttons.primary.disabledBorder')};
```
Comment (colebemis) [inline src/Button/ButtonPrimary.tsx:35]:
```suggestion
    border-color: ${get('colors.btn.primary.disabledBorder')};
```

File: src/Button/ButtonPrimary.tsx [hunk 8]
```diff
@@ -5,32 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    box-shadow: ${get('shadows.btn.primary.selectedShadow')};
+    border-color: ${get('shadows.btn.primary.selectedBorder')};
   }
 
   &:disabled {
-    color: ${get('buttons.primary.color.disabled')};
-    background-color: ${get('buttons.primary.bg.disabled')};
-    border-color: ${get('buttons.primary.border.disabled')};
+    color: ${get('colors.btn.primary.disabledText')};
+    background-color: ${get('buttons.primary.disabledBg')};
```
Comment (colebemis) [inline src/Button/ButtonPrimary.tsx:34]:
```suggestion
    background-color: ${get('colors.btn.primary.disabledBg')};
```

File: src/Button/ButtonOutline.tsx [hunk 9]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    box-shadow: ${get('shadows.btn.outline.selectedShadow')};
+    border-color: ${get('shadows.btn.outline.selectedBorder')};
   }
 
   &:disabled {
-    color: ${get('buttons.outline.color.disabled')};
-    border-color: ${get('buttons.outline.border.default')};
-    background-color: ${get('buttons.outline.bg.disabled')};
+    color: ${get('colors.btn.outline.disabledText')};
+    background-color: ${get('buttons.outline.disabledBg')};
```
Comment (colebemis) [inline src/Button/ButtonOutline.tsx:34]:
```suggestion
    color: ${get('colors.btn.outline.disabledText')};
    background-color: ${get('colors.btn.outline.disabledBg')};
```

File: src/Button/ButtonDanger.tsx [hunk 10]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
-    border-color: ${get('buttons.danger.border.hover')};
-    box-shadow: ${get('buttons.danger.shadow.hover')};
+    color: ${get('colors.btn.danger.hoverText')};
+    background-color: ${get('colors.btn.danger.hoverBg')};
+    border-color: ${get('colors.btn.danger.hoverBorder')};
+    box-shadow: ${get('shadows.btn.danger.hoverShadow')};
   }
   // focus must come before :active so that the active box shadow overrides
   &:focus {
     border-color: transparent;
```
Comment (colebemis) [inline src/Button/ButtonDanger.tsx:21]:
```suggestion
    border-color: ${get('btn.danger.focusBorder')};
```

File: src/Button/ButtonDanger.tsx [hunk 11]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
 
   &:active {
-    color: ${get('buttons.danger.color.active')};
-    background-color: ${get('buttons.danger.bg.active')};
-    box-shadow: ${get('buttons.danger.shadow.active')};
-    border-color: ${get('buttons.danger.border.active')};
+    color: ${get('colors.btn.danger.selectedText')};
+    background-color: ${get('colors.btn.danger.selectedBg')};
+    box-shadow: ${get('shadows.btn.danger.selectedShadow')};
+    border-color: ${get('shadows.btn.danger.selectedBorder')};
```
Comment (colebemis) [inline src/Button/ButtonDanger.tsx:29]:
```suggestion
    border-color: ${get('colors.btn.danger.selectedBorder')};
```

File: src/Button/ButtonDanger.tsx [hunk 12]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
   }
 
   &:disabled {
-    color: ${get('buttons.danger.color.disabled')};
-    background-color: ${get('buttons.danger.bg.disabled')};
-    border: 1px solid ${get('buttons.danger.border.default')};
+    color: ${get('colors.btn.danger.disabledText')};
+    background-color: ${get('buttons.danger.disabledBg')};
```
Comment (colebemis) [inline src/Button/ButtonDanger.tsx:34]:
```suggestion
    background-color: ${get('colors.btn.danger.disabledBg')};
```

File: src/Button/ButtonDanger.tsx [hunk 13]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    border: ${get('buttons.danger.disabledBorder')};
```
Comment (colebemis) [inline src/Button/ButtonDanger.tsx:35]:
```suggestion
    border: ${get('colors.btn.danger.disabledBorder')};
```

File: src/Button/ButtonPrimary.tsx [hunk 14]
```diff
@@ -5,32 +5,34 @@ import {ComponentProps} from '../utils/types'
...
-    box-shadow: ${get('buttons.primary.shadow.focus')};
-    background-color: ${get('buttons.primary.bg.focus')};
+    box-shadow: ${get('colors.btn.primary.focusShadow')};
   }
 
   &:active {
-    background-color: ${get('buttons.primary.bg.active')};
-    box-shadow: ${get('buttons.primary.shadow.active')};
-    border-color: ${get('buttons.primary.border.active')};
+    color: ${get('colors.btn.primary.selectedText')};
```
Comment (colebemis) [inline src/Button/ButtonPrimary.tsx:26]:
```suggestion
```

File: src/Button/ButtonPrimary.tsx [hunk 15]
```diff
@@ -5,32 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    background-color: ${get('colors.btn.primary.selectedBg')};
+    box-shadow: ${get('shadows.btn.primary.selectedShadow')};
+    border-color: ${get('shadows.btn.primary.selectedBorder')};
```
Comment (colebemis) [inline src/Button/ButtonPrimary.tsx:29]:
```suggestion
```

File: src/Button/ButtonClose.tsx [hunk 16]
```diff
@@ -16,11 +16,11 @@ const StyledButton = styled.button<StyledButtonProps>`
   border-radius: ${get('radii.2')};
 
   &:focus {
-    box-shadow: ${get('buttons.close.shadow.focus')};
+    box-shadow: ${get('shadows.btn.shadow')};
```
Comment (colebemis) [inline src/Button/ButtonClose.tsx:19]:
```suggestion
    box-shadow: ${get('shadows.btn.focusShadow')};
```

File: src/Button/ButtonClose.tsx [hunk 17]
```diff
@@ -16,11 +16,11 @@ const StyledButton = styled.button<StyledButtonProps>`
   border-radius: ${get('radii.2')};
```
Comment (colebemis) [inline src/Button/ButtonClose.tsx:16]:
```suggestion
  border-radius: ${get('radii.2')};
  color: ${get('colors.text.secondary')};
```

File: src/Button/ButtonClose.tsx [hunk 18]
```diff
@@ -16,11 +16,11 @@ const StyledButton = styled.button<StyledButtonProps>`
...
 
   &:focus {
-    box-shadow: ${get('buttons.close.shadow.focus')};
+    box-shadow: ${get('shadows.btn.shadow')};
   }
 
   &:active {
-    color: ${get('buttons.close.color.default')};
+    color: ${get('colors.btn.text')};
```
Comment (colebemis) [inline src/Button/ButtonClose.tsx:23]:
```suggestion
    color: ${get('colors.text.link')};
```

File: src/Button/ButtonClose.tsx [hunk 19]
```diff
@@ -16,11 +16,11 @@ const StyledButton = styled.button<StyledButtonProps>`
   border-radius: ${get('radii.2')};
 
   &:focus {
-    box-shadow: ${get('buttons.close.shadow.focus')};
+    box-shadow: ${get('shadows.btn.shadow')};
   }
 
   &:active {
```
Comment (colebemis) [inline src/Button/ButtonClose.tsx:22]:
```suggestion
  &:hover {
```

File: src/Button/ButtonOutline.tsx [hunk 20]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    background-color: ${get('colors.btn.outline.hoverBg')};
+    border-color: ${get('colors.btn.outline.hoverBorder')};
+    box-shadow: ${get('shadows.btn.outline.hoverShadow')};
   }
   // focus must come before :active so that the active box shadow overrides
   &:focus {
-    border-color: transparent;
-    box-shadow: ${get('buttons.outline.shadow.focus')};
+    border-color: ${get('colors.btn.outline.focusBorder')};
+    box-shadow: ${get('shadows..btn.outline.focusShadow')};
```
Comment (colebemis) [inline src/Button/ButtonOutline.tsx:22]:
```suggestion
    box-shadow: ${get('shadows.btn.outline.focusShadow')};
```

File: src/Button/ButtonDanger.tsx [hunk 21]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    border-color: ${get('colors.btn.danger.selectedBorder')};
   }
 
   &:disabled {
-    color: ${get('buttons.danger.color.disabled')};
-    background-color: ${get('buttons.danger.bg.disabled')};
-    border: 1px solid ${get('buttons.danger.border.default')};
+    color: ${get('colors.btn.danger.disabledText')};
+    background-color: ${get('colors.btn.danger.disabledBg')};
+    border: ${get('colors.btn.danger.disabledBorder')};
```
Comment (colebemis) [inline src/Button/ButtonDanger.tsx:35]:
```suggestion
    border-color: ${get('colors.btn.danger.disabledBorder')};
```

File: src/Button/ButtonDanger.tsx [hunk 22]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    color: ${get('colors.btn.danger.hoverText')};
+    background-color: ${get('colors.btn.danger.hoverBg')};
+    border-color: ${get('colors.btn.danger.hoverBorder')};
+    box-shadow: ${get('shadows.btn.danger.hoverShadow')};
   }
   // focus must come before :active so that the active box shadow overrides
   &:focus {
-    border-color: transparent;
-    box-shadow: ${get('buttons.danger.shadow.focus')};
+    border-color: ${get('btn.danger.focusBorder')};
```
Comment (colebemis) [inline src/Button/ButtonDanger.tsx:21]:
```suggestion
    border-color: ${get('colors.btn.danger.focusBorder')};
```

File: src/Button/ButtonDanger.tsx [hunk 23]
```diff
@@ -5,34 +5,34 @@ import {ComponentProps} from '../utils/types'
...
+    box-shadow: ${get('colors.btn.danger.focusShadow')};
```
Comment (colebemis) [inline src/Button/ButtonDanger.tsx:22]:
```suggestion
    box-shadow: ${get('shadows.btn.danger.focusShadow')};
```

**Review by colebemis — APPROVED** [review]
Awesome! Looks like you just need to update the snapshots and this should be ready to merge 🚢

**Pull request merged**